core/iwasm/common/wasm_runtime_common.c

```diff
@@ -871,6 +871,25 @@ get_package_type(const uint8 *buf, uint32 size)
     return Package_Type_Unknown;
 }
 
+unsigned int
+wasm_runtime_module_get_package_version(wasm_module_t module)
```

Comment:
No need to add this function if #3600 is merged, we can use
```C
    package_type_t package_type = get_module_package_type(module);
    uint32 package_current_version = get_package_current_version(package_type);
```
instead.

Reply:
I agree that this would work for existing WAMR implementations, which only support a single version of WASM files, and a single version of AOT files. When I created the API, I was also considering the possibility that some future WAMR implementation might support multiple versions of either one of these types of files. Do you think it's worth planning for that possibility in this API, or do you think if for example WASM version 2 is released that WAMR would stop supporting WASM version 1?

Reply:
For AOT, normally the AOT runtime only supports the latest AOT file format, or AOT_CURRENT_VERSION, it is difficult to support backward compatibility since the AOT file format may change a lot from one version to another version. For interpreter, I am not very sure but it may be possible to support both two versions in runtime, since normally there is a good backward compatibility in the wasm file format.

Here I think maybe we want to get: (1) the version in the wasm/aot file, (2) the version of a wasm/aot module, (3) the current version that runtime supports.
For (1), how about:
```C
uint32_t wasm_runtime_get_file_package_version(const uint8_t *buf, uint32_t size);
```

For (2), I think we should add a field (e.g. `uint32 package_version`) in WASMModule/AOTModule to save the version loaded and return it in the API, how about adding the api:
```C
uint32_t wasm_runtime_get_module_package_version(const wasm_module_t module);
```

For (3), how about
```C
uint32_t wasm_runtime_get_current_package_version(package_type_t package_type);

or

uint32_t wasm_runtime_get_max_package_version(package_type_t package_type);
```
which returns AOT_CURRENT_VERSION or WASM_CURRENT_VERSION. But I am not sure which API name is better.

How do you think?